portfolio/src/main/webapp/script.js

```diff
@@ -13,16 +13,52 @@
 // limitations under the License.
 
 /**
- * Adds a random greeting to the page.
+ * Gets random valid index of an given array
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+function getRandomIndex(array) {
+  return Math.floor(Math.random() * array.length);
+}
+
+/**
+ * Displays the text in the element in the page with given id
+ */
+function addTexttoPage(text, id){
+  const container = document.getElementById(id);
+  container.innerText = text;
+}
+
+// Global variable to track which topic has been randomly selected. 
+// Initialize to -1 to represent no topic is selected when page is first loaded. 
+var topicIndex = -1;
+const books =
+  ['Rangers', 'Exhalation', 'The Elegant Universe', 'Human Acts'];
+const tvshows =
+  ['Brookyln Nine Nine', 'Designated Survivor', 'Person of Interest', 'The Good Place', 'Broadchurch'];
+const korean =
+  ['안녕하세요!', '좋은 하루 되세요!', '오늘 날씨는 어떤가요?', '졸려요', '고맙습니다'];
+const misc =
+  ['Penguin', 'Blue', 'Baseball', 'Cheesecake'];
+const cumultative =
```

Comment:
nit: can you do something like this:

```
const TOPIC_ITEMS = {
   'book recommendation': ['Rangers', .... ]
   'tv show recommendation': ['Brooklyn Nine Nine', ...]
};
```
You can then get `topics` with [Object.keys](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys). You can index an object the same way as an array, but with strings instead of numbers.

This will have less potential for mistakes (e.g. in the original code, you have to make sure `cumulative` matches with `topics` exactly).

As a challenge, you can also try using a [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) instead of an object. You can use the [example](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/Map) to initialize the Map in one go.